src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -101,6 +103,8 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 		return false; //same formula already set - no change
 
 	//really new formula - set new FORMULA
+	if (!_quantity->GetFormula())
+		switchJacobians = true;
```

Comment:
This is wrong. In this case, **switchJacobians** must be set to true if either the **old** formula or the **new** formula depends on any state variable.